jwplatform/client.py

```diff
@@ -96,12 +97,48 @@ def request(self, method, path, body=None, headers=None, query_params=None):
 
 class _ScopedClient:
 
-    def __init__(self, client):
+    def __init__(self, client: JWPlatformClient):
         self._client = client
 
 
-class _ResourceClient(_ScopedClient):
+class _GenericResourceClient(_ScopedClient):
+    _resource_name = None
+    _id_name = None
+    _collection_path = "/v2/{resource_name}/{resource_id}/{subresource_name}"
+    _singular_path = "/v2/{resource_name}/{resource_id}/{subresource_name}/{subresource_id}"
+
+    def __init__(self, client: JWPlatformClient):
+        super().__init__(client)
+
+    def list(self, resource_name, resource_id, subresource_name, query_params=None):
+        response = self._client.request(
+            method="GET",
+            path=self._collection_path.format(resource_name=resource_name, resource_id=resource_id,
+                                              subresource_name=subresource_name),
+            query_params=query_params
+        )
+        return ResourcesResponse.from_client(response, subresource_name, self.__class__)
+
 
+class _UploadClient(_GenericResourceClient):
+    _collection_path = "/v1/uploads/{resource_id}/{subresource_name}"
+    _singular_path = "/v1/uploads/{resource_id}/{subresource_name}/{subresource_id}"
+
+    def __init__(self, api_secret, base_url='upload.jwplayer.com'):
+        client = JWPlatformClient(secret=api_secret, host=base_url)
+        super().__init__(client)
+
+    def complete(self, resource_name, resource_id, subresource_name, body=None):
```

Comment:
I think it may be unnecessary to take in `resource_name` and `subresource_name` here, since you know they should always be "uploads" and "complete"